Add rake task to report on commonly used passwords

brucebolt opened the pull request:
We have [recently implemented](https://github.com/alphagov/govuk-account-manager-prototype/pull/495) a block on new users using a password on the NCSC's commonly used password list.

However we believe existing users may have already used these passwords. This task will report how many users this affects, so we can plan on how to ask them to change their password.

Since the task is slow and could be terminated by a deployment, it is run as a Sidekiq job in a separate queue (with the lowest priority), so as not to block any other queued tasks. If re-run, it will only check users who do not have an existing check result stored.

Trello card: https://trello.com/c/N4Yzytap

barrucadu commented on lib/tasks/security.rake:
```diff
@@ -1,7 +1,29 @@
+  desc "Report on usage of passwords on NCSC's password denylist."
+  task report_ncsc_denylist_usage: :environment do
+    User.where(ncsc_password_match: nil).each do |user|
+      user_hashed_password = user.encrypted_password # pragma: allowlist secrets
+      bcrypt = ::BCrypt::Password.new(user_hashed_password)
+
+      hashed_nscs_password_list = BannedPassword.pluck(:password).map do |password|
+        if User.pepper.present?
+          password = "#{password}#{User.pepper}" # pragma: allowlist secrets
+        end
+        ::BCrypt::Engine.hash_secret(password, bcrypt.salt)
+      end
+
+      user.update!(ncsc_password_match: hashed_nscs_password_list.include?(user_hashed_password))
```
This could be fairly slow as we're effectively constructing a rainbow table for each user and then checking if we get a hit.

I think this might be a bit faster, as it likely won't need to hash *all* the banned passwords if there is a hit:

```ruby
has_banned_password = BannedPassword.pluck(:password).any? do |password|
  User. valid_password?(password)
end

user.update!(ncsc_password_match: has_banned_password)
```

It also means we don't need to know the details of how devise hashes passwords.

brucebolt replied:
Thanks!  This does seem considerably faster and has simplified the code.

barrucadu reviewed (commented):
Nit: the job and the task don't really use the NCSC list; they use the list in our database.  Which currently will be the NCSC list.  But this change will work equally well if we used some other source of banned passwords.

Maybe rename things to something like `UserBannedPasswordCheckJob` and `check_banned_password_usage`?

barrucadu commented on app/jobs/ncsc_user_password_check_job.rb:
```diff
@@ -0,0 +1,13 @@
+class NcscUserPasswordCheckJob < ApplicationJob
```
It would be nice to have a test for this job

brucebolt commented:
Great idea, I've renamed everything from NCSC to Banned.  Also added a test.

Had to disable a cop in the tests (`SkipsModelValidations`) since we actually need to skip model validations in order to set a password that is on the banned list.
